docs/custom-plugins/visitor.md

```diff
@@ -0,0 +1,111 @@
+# Visitor pattern
+
+_Visitor_ is a design pattern that allows operations to be performed on individual elements in a more complex structure. Redocly uses it as the basis for custom plugins where every node in a document must be evaluated, for example when applying rules or decorators.
+
+To understand how this applies to your API description, think of the document as a tree structure. The top level elements are entries like `info` and `components`. To examine the `description` field in the `info` section, the visitor goes to the `info` node first, then on to the `description` node. This pattern is repeated all over the document as the visitor pattern works its way around the document tree.
+
+<a id="visitor-object"></a><a id="format-of-visitor-object"></a>
```

Comment:
```suggestion
```
Suggestion: Remove -- I didn't find any anchor links to either of these anchors.